### PR TITLE
Add command for announcing new article

### PR DESCRIPTION
**Pull request by adrianm27r**
adding new ```announce``` command

### CODE REVIEW

File: src/commands/check.js [hunk 1]
```diff
@@ -7,13 +7,14 @@ import {owner} from '../config/config.json';
...
 
-let url = 'https://www.craneanime.com/wp/feed/atom';
+let url = 'https://xkcd.com/atom.xml';
 
 export default {
   name: 'check',
   description: 'Check the atom feed',
   guildOnly: true,
   execute(message: Message) {
+    // console.log('message', message);
```
Comment (darcien) [inline src/commands/check.js]: Nit: can we remove unused console log?

File: src/commands/check.js [hunk 2]
```diff
@@ -28,8 +29,9 @@ export default {
 
     checkFeed(url).then(async(result) => {
       let newArticles = [];
-
+      console.log('result', result);
```
Comment (darcien) [inline src/commands/check.js]: Same as above.

File: src/commands/index.js [hunk 3]
```diff
@@ -3,3 +3,4 @@
 export {default as about} from './about';
 export {default as check} from './check';
 export {default as ping} from './ping';
+export {default as postnsave} from './postnsave';
```
Comment (darcien) [inline src/commands/index.js]: Nit: since post or announcing stuffs by default will be saving them anyway, can we rename the command? I think `announce` is more appropriate

File: src/commands/postnsave.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+let url = 'https://www.craneanime.com/wp/feed/atom';
+// let url = 'https://xkcd.com/atom.xml';
+
+export default {
+  name: 'postnsave',
+  description: 'Check the atom feed',
+  guildOnly: true,
+  execute(message: Message, args: String) {
+    // console.log('message', message.guild.id);
```
Comment (darcien) [inline src/commands/postnsave.js]: Same, please remove unused log.

File: src/commands/postnsave.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+import type {Message} from 'discord.js';
+import type {Stash} from '../type/Stash';
+
+let url = 'https://www.craneanime.com/wp/feed/atom';
+// let url = 'https://xkcd.com/atom.xml';
+
+export default {
+  name: 'postnsave',
+  description: 'Check the atom feed',
```
Comment (darcien) [inline src/commands/postnsave.js]: Should be `Check the feed and make announcement for new post`?

File: src/commands/postnsave.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  guildOnly: true,
+  execute(message: Message, args: String) {
+    // console.log('message', message.guild.id);
+    let newArticles = [];
+    if (message.author.id !== owner) {
+      return;
+    }
+
+    message.channel.send('Checking feed...').then((sentMessage) => {
+      if (Array.isArray(sentMessage)) {
```
Comment (darcien) [inline src/commands/postnsave.js]: Why do we return if `sentMessage` is an array?

File: src/commands/postnsave.js [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        return;
+      }
+
+      sentMessage.delete(3000);
+    });
+
+    checkFeed(url).then(async(result) => {
```
Comment (darcien) [inline src/commands/postnsave.js]: You don't need `async` if you don't plan to use `await`.

File: src/config/exampleConfig.json [hunk 8]
```diff
@@ -1,5 +0,0 @@
-{
```
Comment (darcien) [inline src/config/exampleConfig.json]: Why did you remove the example config...
Did you change it to your config without copying the file?

File: src/commands/postnsave.js [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+          //utf8 encoding change data to string
+          fs.readFile(storagePath, 'utf8', (err, data) => {
+            if (err) {
+              console.log(err);
+            }
+
+            let parsedArticleData: Stash = JSON.parse(data);
+
+            let articlesCheck = newArticles.filter((article) => {
+              return article.guid === parsedArticleData.articles[0].guid;
```
Comment (darcien) [inline src/commands/postnsave.js]: Why is it just comparing with the first article guid?

Oh well, for now, just leave it like that.

File: src/commands/postnsave.js [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+          let obj;
+          //utf8 encoding change data to string
+          fs.readFile(storagePath, 'utf8', (err, data) => {
+            if (err) {
+              console.log(err);
+            }
+
+            let parsedArticleData: Stash = JSON.parse(data);
+
+            let articlesCheck = newArticles.filter((article) => {
```
Comment (darcien) [inline src/commands/postnsave.js]: I think this should be `filteredArticles` instead

File: src/commands/postnsave.js [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+        if (err == null) {
+          //file exist
+          let obj;
+          //utf8 encoding change data to string
+          fs.readFile(storagePath, 'utf8', (err, data) => {
+            if (err) {
+              console.log(err);
+            }
+
+            let parsedArticleData: Stash = JSON.parse(data);
```
Comment (darcien) [inline src/commands/postnsave.js]: So, is it a Stash or article data?

Comment (darcien) [reply src/commands/postnsave.js]: Let's pick one, since you already call it stash, let's use `parsedStash`

File: src/commands/postnsave.js [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+            let articlesCheck = newArticles.filter((article) => {
+              return article.guid === parsedArticleData.articles[0].guid;
+            });
+            if (articlesCheck.length === 0) {
+              for (let article of newArticles) {
+                parsedArticleData.articles.push(article);
+              }
+              let articleStringify = JSON.stringify(parsedArticleData, null, 2);
```
Comment (darcien) [inline src/commands/postnsave.js]: Stringify means to turn something into a string, I think you can use `newStashString` or something along that line. but we can do that later.

File: src/commands/postnsave.js [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+
+              fs.writeFile(storagePath, articleStringify, (err) => {
+                if (err) {
+                  message.channel.send('Fail to save this');
```
Comment (darcien) [inline src/commands/postnsave.js]: Saving the article is related to the devs only, so I don't think we should send it to the channel.

File: src/commands/postnsave.js [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+                }
+                message.channel.send('Feeds is updated in storage');
```
Comment (darcien) [inline src/commands/postnsave.js]: Same as above.

File: src/commands/postnsave.js [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+              });
+            } else {
+              message.channel.send('articles already in stash');
```
Comment (darcien) [inline src/commands/postnsave.js]: Sometimes it's storage, sometimes it's stash 😆

**Review by darcien — APPROVED** [review]
Could you please ignore `.DS_Store` file with your global gitignore, after that, please remove it from your commit.

After that's done, I'll merge this and patch it up ya.

File: src/commands/postnsave.js (hunk 6 above)
Comment (adrianm27r) [reply src/commands/postnsave.js]: this one comes from the ```check``` command. ill just remove it then sorry 😄

Comment (darcien) [reply src/commands/postnsave.js]: Oops, that means I totally forgot about that.

File: src/config/exampleConfig.json (hunk 8 above)
Comment (adrianm27r) [reply src/config/exampleConfig.json]: yeah i changed it to config, will revert it back

File: src/commands/index.js (hunk 3 above)
Comment (adrianm27r) [reply src/commands/index.js]: yeah, sorry about the lack of creativity on the naming 🤣. will change it

File: src/commands/postnsave.js (hunk 5 above)
Comment (adrianm27r) [reply src/commands/postnsave.js]: oh i think i skipped that, will change it

File: src/commands/postnsave.js (hunk 7 above)
Comment (adrianm27r) [reply src/commands/postnsave.js]: oh yeah, wew bad habit of using but not analysing through 😸
